buf/registry/module/v1beta1/resource.proto

```diff
@@ -43,24 +42,20 @@ message ResourceRef {
   //
   // The following semantics are applied:
   //   - If the child oneof is not specified, the name is interpreted to reference a Module.
-  //   - If branch_name is specified, the name is interpreted to reference a Branch.
-  //   - If tag_name is specified, the name is interpreted to reference a Tag.
-  //   - If vcs_commit_hash is specified, the name is interpreted to reference a VCSCommit.
+  //   - If label_name is specified, the name is interpreted to reference a Label.
   //   - If digest is specified, the name is interpreted to reference a Commit with this Digest.
   //   - If ref is specified, it is interpreted to be either an id or name.
   //     - If an id, this is equivalent to setting the id field on ResourceRef. However,
   //       backends can choose to validate that the owner and module fields match the resource
   //       referenced, as additional validation.
-  //     - If a name, this is interpreted to be either a Branch name, Tag name,
-  //       VCSCommit hash, or Digest string. A digest string is of the form "typeString:hexValue",
-  //       where typeString is the lowercase of the Type value with TYPE_ stripped, and hexValue
-  //       is the hex-encoded value of the digest.
-  //     - If there is a conflict between names across resources (for example, there is a
-  //       Branch and Tag with the same name), the following order of precedence is applied:
+  //     - If a name, this is interpreted to be either a Label name or Digest string. A Digest string
+  //       is of the form "typeString:hexValue", where typeString is the lowercase of the Type
+  //       value with TYPE_ stripped, and hexValue is the hex-encoded value of the digest.
+  //     - If there is a conflict between names across resources (for example, there is a Commit id
+  //       and Label name of the same value), the following order of precedence is applied:
   //       - Commit
-  //       - VCSCommit
-  //       - Tag
-  //       - Branch
+  //       - Label
+  //       - Digest
```

Comment:
I would think Digest needs to have precedence over a label. The whole point of a digest value is it always points to the same content, but if I can create a label that redefines a digest value, that is kind of like a hash collision attack.

Reply:
Done